siliconcompiler/core.py

```diff
@@ -2525,13 +2524,12 @@ def __archive_job(self, tar, job, steplist, index=None, include=None):
         else:
             self.logger.warning('Archiving job with failed or incomplete run.')
 
-        indexlist = [index] if index else None
-        for (step, idx) in self._get_flowgraph_nodes(flow, steplist=steplist, indexlist=indexlist):
+        for (step, idx) in flowgraph_nodes:
             self.logger.info(f'Archiving {step}{idx}...')
             self._archive_node(tar, step, idx, include=include)
 
     ###########################################################################
-    def archive(self, jobs=None, step=None, index=None, include=None, archive_name=None):
+    def archive(self, jobs=None, include=None, archive_name=None):
```

Comment:
this should not change, if step and or index is provided, only those should be archived.